tsec-http4s/src/main/scala/tsec/authentication/JWKSAuthenticator.scala

```diff
@@ -0,0 +1,185 @@
+package tsec.authentication
+
+import java.math.BigInteger
+import java.time.{Instant, LocalDateTime}
+import java.util.Base64
+
+import cats.data.OptionT
+import cats.effect.{Effect, Sync}
+import cats.syntax.all._
+import io.circe.Decoder
+import io.circe.parser.decode
+import org.http4s.headers.Authorization
+import org.http4s.{AuthScheme, Credentials, EntityDecoder, Request, Response, Uri}
+import tsec.common.SecureRandomId
+import tsec.jws.signature.{JWSSigCV, JWTSig}
+import tsec.signature.jca.{JCASigTag, SigPublicKey}
+
+import scala.concurrent.duration.FiniteDuration
+
+final case class AugmentedJWKS[A, I](
+    id: SecureRandomId,
+    jwt: JWTSig[A],
+    identity: I,
+    expiry: Instant,
+    lastTouched: Option[Instant]
+)
+
+final case class JWK(kid: String, alg: String, kty: String, use: String, n: String, e: String)
```

Comment:
Notes:
- JWK are simply a key representation format. Your "JWK Authenticator", as it is written, is a "JWK using an RSA public key authenticator". I'm ok with adding a specific use case (since you wrote it), but given this is definitely not generic, we either make it support the JWK spec for other things by making JWKs more generic, or we change this to state its actual intent
- This only seems to support RSA. Why?
- given this is only for RSA, why is `alg` not a fixed field?
- `n` and `e`  are public key parameters for RSA. This does not support RSA private keys.

Reply:
Moreover looking down, it seems you're using n and e as b64 encoded strings. I'd prefer newtypes for this if possible.